### ISSUE TITLE
Make issue warning clearer when offering to add `brand:wikidata` or `brand:wikipedia`

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/iD#6517]
Whole point of `brand:wikidata` and `brand:wikipedia` is that name alone (or combined with other indicator like object type and location) is not sufficient to identify object as belonging to a specific brand.

Despite that iD is asking user to add `brand:wikidata` and `brand:wikipedia` and describes this action as "XYZ looks like a brand with nonstandard tags" and offers user to "Upgrade the tags".

There are multiple problems here

- iD is adding new tags, not "upgrading" ones that are present, so description is misleading
- such heuristic usually gives good results, but there are some false positives. iD must mention to user that he is responsible for verifying this and that suggested change may be wrong
- what is actually changed should be displayed in form readable to user
- Even after clicking "i" button this tags are still not linking to Wikipedia/Wikidata making verification harder

See for example http://preview.ideditor.com/master/#background=Bing&disable_features=boundaries&id=w488118123&map=20.29/38.74517/-75.15497

![Selection_004](https://user-images.githubusercontent.com/899988/59211598-8269fd80-8bb0-11e9-91b7-f83a06bf782c.png)

After #6443 it was improved a bit but is still far from OK version.

Note also that

> All features of the same brand should be tagged the same way.

displayed after clicking "i" is wrong - for example shops of the same brand may have different `opening_hours` or other tags.

**Comment by quincylvania** [comment]
> Whole point of brand:wikidata and brand:wikipedia is that name alone (or combined with other indicator like object type and location) is not sufficient to identify object as belonging to a specific brand.
>
> Despite that iD is asking user to add brand:wikidata and brand:wikipedia and describes this action as "XYZ looks like a brand with nonstandard tags" and offers user to "Upgrade the tags".

The message says "looks like", which means that iD isn't sure. The user is still making their own judgement. "Ignore this issue" is an option.

> iD is adding new tags, not "upgrading" ones that are present, so description is misleading

Upgrading means iD will update to the preferred tagging for the same thing. Whether this involves adding, removing, or changing tags isn't really a factor.

> such heuristic usually gives good results, but there are some false positives. iD must mention to user that he is responsible for verifying this and that suggested change may be wrong

Maybe. The wording doesn't imply absolute certainty and the user is always responsible for their edits anyway.

> what is actually changed should be displayed in form readable to user

It is when you click the info button. We're not going to show this all the time since it takes up a lot of space.

>> All features of the same brand should be tagged the same way.

> displayed after clicking "i" is wrong - for example shops of the same brand may have different `opening_hours` or other tags.

The message doesn't mean that all the tags should be identical. It's like answering "how should I tag a new Walmart?" A new Walmart should be tagged the same as an old Walmart. If this wording causes real problems we can change it.

**Comment by matkoniecz** [comment]
>  The user is still making their own judgement. "Ignore this issue" is an option.

"Upgrade the tags" vs "Ignore the issue" at least for me sounds like, "are you willing to fix this issue", not "please check is it correct".

Note that it is combined with (by default) hiding what will actually happen.

> Upgrading means iD will update to the preferred tagging for the same thing. Whether this involves adding, removing, or changing tags isn't really a factor.

My problem here was "XYZ looks like a brand with nonstandard tags" message - there are no "nonstandard tags" here what is even more confusing. "XYZ looks like a brand where tags may be added" would be better.

For cases where iD changes/removes existing tags "XYZ looks like a brand where tags may be changed" would be better.

**Comment by matkoniecz** [comment]
In general, I think that it should be "Mark as brand XYZ {Brand XYZ logo image} {link to wikipedia/wikidata page}" to allow proper verification by tagging user.

**Comment by 1ec5** [comment]
> The message says "looks like", which means that iD isn't sure. The user is still making their own judgement. "Ignore this issue" is an option.

“Looks like” is the only thing about this warning that equivocates. A typical user unfamiliar with the purpose of the QIDs will see a yellow background, ⚠️ icon, and the negative words “issue” and “nonstandard” and conclude that ignoring isn’t the right thing to do.

The upgrade message is intended to merely a helpful suggestion, so it should look informational, not alerting. For example, a light blue background, ℹ️ icon, and even a “Did you mean?” kind of message would go a long way toward ensuring an unbiased appearance.

**Comment by matkoniecz** [comment]
To be more specific, here is my mockup of solution that

- presents change as likely to be valid, but not guaranteed to be correct
- is not misleadingly claiming that current tagging is "nonstandard"
- allows typical user to verify what clicking on button will do

![Selection_006](https://user-images.githubusercontent.com/899988/59293550-afd1ac80-8c7f-11e9-8b5d-dd813d53824f.png)

For comparison current behavior:

![Selection_004](https://user-images.githubusercontent.com/899988/59211598-8269fd80-8bb0-11e9-91b7-f83a06bf782c.png)

**Comment by 1ec5** [comment]
> ⚠️ Mark this object as operated uder brand<br>described at [Citizens Bank (South Bend,<br>Indiana)](https://en.wikipedia.org/wiki/Citizens_Bank_%28South_Bend,_Indiana%29) Wikipedia page?

I find this wording a bit confusing too. Without knowing the purpose of this warning, the fact that it’s presented as a warning would probably still lead an inexperienced mapper to upgrade the feature without checking. A more descriptive message could potentially overcome the warning-like presentation:

> <img src="https://user-images.githubusercontent.com/1231218/59294807-239b9700-8c37-11e9-8363-4c23ecce6cdf.png" width="48" alt="" align="left"> Does this feature belong<br>to the [Noah's Bagels](https://www.wikidata.org/wiki/Q64517373) brand?

**Comment by matkoniecz** [comment]
Yeah, that would be even better.

**Comment by CloCkWeRX** [comment]
> Does this feature belong to the Noah's Bagels brand?

A related issue that might be worth revisiting; on selection of the wikidata brand via a preset, the name is often overridden; then locked.

There seem to be two styles of "brand" vs "name" - for example, a car dealership which is branded with Holden (https://www.wikidata.org/wiki/Q29281) but named "Duttons Holden":
![image](https://user-images.githubusercontent.com/365751/59317686-21067380-8d03-11e9-9fa8-c0247a84040b.png)

This is likely because the dealer is independent, but has signed up to a franchise or similar; vs more "corporate owned" stores; which rely on strictly the branding:

![image](https://user-images.githubusercontent.com/365751/59317880-0c76ab00-8d04-11e9-9b7d-8bcc3fd65cbd.png)

... even if internally or on websites, they refer to themselves as some variant of Brand + Location - ie - https://www.coles.com.au/store-locator/stores/sa/firle-401 and https://www.kmart.com.au/store-detail/Firle

It might be worth relaxing the locking of name on setting of brand:wikidata; or otherwise loosening the relationship between brand + name autocorrection.

**Comment by matkoniecz** [comment]
@CloCkWeRX It is a related but separate issue - can you consider moving it into a new issue (if not yet reported)?

**Comment by hfs** [comment]
A similar situation arises when the name is not unique and there are several potential matches in the NSI.

In Germany we have the situation that there are two supermarket chains with three brands all colloquially called “Netto”.

* [Netto Marken-Discount](https://nsi.guide/index.html?t=brands&k=shop&v=supermarket&tt=netto#nettomarkendiscount-0a839e), often only gets tagged as `name=Netto`,
* [Netto City](https://nsi.guide/index.html?t=brands&k=shop&v=supermarket&tt=netto#nettocity-0a839e) is a separate brand of Netto Marken-Discount for small shops, and
* [Netto (Salling)](https://nsi.guide/index.html?t=brands&k=shop&v=supermarket&tt=netto#netto-bc67ba) where `name=Netto` and `brand=Netto` are correct.

If you edit a supermarket tagged `name=Netto`, e.g. https://www.openstreetmap.org/edit?editor=id&way=31727346#map=19/52.65117/13.19552, it suggests to brand it as Netto (Salling).

![image](https://github.com/openstreetmap/iD/assets/538109/1cccbcaf-dcd1-4c49-996a-27a14da26266)

Instead, it would be better if it would recognize that there are several potential matches in the NSI and offer all of them with logos. This would make it clear that the user has to make a choice and that iD can’t know which brand is the right one.

When adding a *new* node and searching for `Netto` iD lists the brands with logos (if enabled):

![image](https://github.com/openstreetmap/iD/assets/538109/4be09a80-4c7e-41ed-b2b6-d09897bdf19c)

That makes it much more likely that a correct choice is made. I imagine that a similar choice could be displayed for the case of unambiguous brands, with the additional choice of “none of these”.

**Comment by ginnyTheCat** [comment]
Additionally there is to mention that Netto (Salling) is way smaller than Netto Marken-Discount, and many people don't even know it exists here in Germany.

**Comment by matkoniecz** [comment]
Thanks for https://github.com/Openstreetmap/iD/commit/88f106ebd92bd1596f27aa21dc119ecba0180355 !
